roles/launch_vms/files/upload_image

```diff
@@ -0,0 +1,159 @@
+        wait_for_disk(disks_service, disk)
+    except Exception as e:
+        if '409' not in ' '.join(e.args):
+            raise
+
+
+def get_resource(service, name):
+    try:
+        resource = service.list(search='name={}'.format(name))
+    except TypeError:
+        # Not all resource types allow the search parameter
+        # In this case, will have to manually filter
+        resource = list(filter(lambda x: x.name == name, service.list()))
+    return resource[0] if resource else None
+
+
+def create_disk(disks_service, disk_name, disk_size, storage_domain):
+    return get_resource(disks_service, disk_name) or disks_service.add(
+        types.Disk(
+            name=disk_name,
+            bootable=True,
+            active=True,
+            description='Template disk for creating rhel7 VMs',
+            format=types.DiskFormat.COW,
+            provisioned_size=disk_size * 2**30,
+            storage_domains=[
+                types.StorageDomain(
+                    name=storage_domain
+                )
+            ]
+        )
+    )
+
+
+def wait_for_disk(disks_service, disk):
+    disk_service = disks_service.disk_service(disk.id)
+    while disk.status != types.DiskStatus.OK:
+        time.sleep(5)
+        disk = disk_service.get()
+    return disk
+
+
+def transfer_image(transfers_service, disk_id, image_path):
+    transfer = transfers_service.add(
+        types.ImageTransfer(
+            image=types.Image(
+                id=disk_id
+            )
+        )
+    )
+
+    transfer_service = transfers_service.image_transfer_service(transfer.id)
+    while transfer.phase == types.ImageTransferPhase.INITIALIZING:
+        time.sleep(3)
+        transfer = transfer_service.get()
+
+    upload_headers = {
+        'Authorization':  transfer.signed_ticket
+    }
+
+    # Extract the URI, port, and path from the transfer's proxy_url.
+    url = urlparse(transfer.proxy_url)
+
+    # Create a new instance of the connection.
+    proxy_connection = HTTPSConnection(
+        url.hostname,
+        url.port,
+        context=ssl.SSLContext(ssl.PROTOCOL_SSLv23)
+    )
+
+    bytes_header = "bytes {}-{}/{}".format
+    with open(image_path, "rb") as disk:
+        size = os.path.getsize(image_path)
+        chunk_size = 1024*1024*MB_per_request
+        pos = 0
+        while (pos < size):
+            transfer_service.extend()
+
+            upload_headers['Content-Range'] = bytes_header(
+                pos,
+                min(pos + chunk_size, size)-1,
+                size
+            )
+            proxy_connection.request(
+                'PUT',
+                url.path,
+                disk.read(chunk_size),
+                headers=upload_headers
+            )
+            r = proxy_connection.getresponse()
+            progress = pos / float(size)
+            print(r.status, r.reason, "Progress: ", "{:.0%}".format(progress))
+            pos += chunk_size
+
+    transfer_service.finalize()
```

Comment:
You should check if it succed, but this operation support ovirt_disks modules so no need to do it here.